chain/src/main/scala/org/bitcoins/chain/blockchain/ChainHandler.scala

```diff
@@ -89,3 +95,24 @@ case class ChainHandler(
     }
   }
 }
+
+object ChainHandler {
+
+  def apply(blockHeaderDAO: BlockHeaderDAO, chainConfig: ChainAppConfig)(
```

Comment:
So the idea is to provide one constructor for DB access (returning a future), and one without DB access returning instantly? I think that's a good idea. Although it might make sense to give them more descriptive names, lately I've been leaning towards that having a bunch of `apply` methods can be confusing and overwhelming

Reply:
No, the other constructor is simply a convinence method for passing in one `Blockchain`. It simply does `Vector(blockchain)` and then calls the default case class constructor. I suppose I could use some sort of vargs here but I'm unsure about what collection it defaults to and what the performance implications are.

Reply:
Ok, I see. What's your thoughts on renaming the first `apply`, to something like `fromDatabase`? To signify that its querying the DB for the current state